### ISSUE TITLE
Websocket definition in groups

### ISSUE THREAD

**Issue by ntzrbtr** [elysiajs/elysia#27]
When using groups, you can't use the ws() method on the app as inside the group, "app" is a fresh instance of Elysia. So to define a websocket endpoint inside of a group, you have to add the plugin once more to the group instance.

This does not work:

``` typescript
import { Elysia, ws } from "elysia";

const app = new Elysia()
	.use(ws())
	.get("/", () => "Welcome to Elysia!")
	.group("/group", (app) => app.ws("/websocket", {})
	.listen(3000);
```

This works:

``` typescript
import { Elysia, ws } from "elysia";

const app = new Elysia()
	.use(ws())
	.get("/", () => "Welcome to Elysia!")
	.group("/group", (app) => app.use(ws()).ws("/websocket", {})
	.listen(3000);
```

Is this intended as such? If yes, I would try to clarify that in the docs.

**Comment by SaltyAom** [comment]
Thanks for the report, from a technical perspective, it's expected.

However, I think your approach should work by default too, and I update Elysia 0.3.2 to work with the first snippet.

Should be fixed with #f313078, please update to 0.3.2 to verify if your first snippet work.

**Comment by ntzrbtr** [comment]
I worked on my project further last night and somehow the websocket connection could not be established when it was defined inside the group. So now I moved it outside and it is working.

So from

```typescript
import { Elysia, ws } from "elysia";

const app = new Elysia()
	.use(ws())
	.get("/", () => "Welcome to Elysia!")
	.group("/group", (app) => app.get("/", myHandler).ws("/websocket", {}))
	.listen(3000);
```

I moved to

```typescript
import { Elysia, ws } from "elysia";

const app = new Elysia()
	.use(ws())
	.get("/", () => "Welcome to Elysia!")
	.group("/group", (app) => app.get("/", myHandler))
	.ws("/group/websocket", {})
	.listen(3000);
```

**Comment by SaltyAom** [comment]
Yes, that's expected. And I'm trying to make `ws` inside `group` work too.

With the patch of Elysia 0.3.2, this code should also work:
```ts
import { Elysia, ws } from "elysia";

const app = new Elysia()
	.use(ws())
	.get("/", () => "Welcome to Elysia!")
	.group("/group", (app) => app.get("/", myHandler).ws("/websocket", {}))
	.listen(3000);
```

Can you verify that the code above work on your end?

**Comment by ntzrbtr** [comment]
Sorry, I wasn't working on my project for some days.

Should I test this again with v0.3.2 or directly with the latest v0.4.5?

**Comment by SaltyAom** [comment]
0.4.5 should work

**Comment by ntzrbtr** [comment]
OK, I tested this extensively and when defining the websocket route inside a group, the connection can't be upgraded. I tried to debug this, but I can't see a reason for the upgrade failure, as Bun just returns a boolean when calling server.upgrade().

**Comment by vandelli** [comment]
+1 same issue here with version `0.7.15`
Unable to connect to ws after using it under group.

**Comment by Blechlawine** [comment]
I think I'm having the same issue. When I register a websocket handler in a group, like this:
```typescript
import Elysia from "elysia";

new Elysia()
    .group("/ws", (a) =>
        a.ws("/", {
            open() {
                console.log("Hello");
            },
        }),
    )
    .listen(3000);
```
When trying to connect to it, i always get an error code 400 with this message:
`Expected a websocket connection`

I think it might have something to do with `server` when upgrading the request. Maybe `server` ist undefined | null, or another falsy value, thus the if upgrade fails, and this error is returned? (relevant code: https://github.com/elysiajs/elysia/blob/4a1134b082b726eb67b0581565628a1d697a0c18/src/index.ts#L2401C5-L2402C45)

**Comment by SaltyAom** [comment]
Fix with eabe264, available on 0.7.27.

Feel free to reopen if the issue still persists.
